src/utils.ts

```diff
@@ -31,3 +32,36 @@ export const getDownloadUrl = async (version: string): Promise<string> => {
   }
   return `https://github.com/supabase/cli/releases/download/v${version}/${filename}`
 }
+
+export const resolveRedirect = async (url: string): Promise<string> => {
```

Comment:
On a second thought, using https://github.com/actions/toolkit/tree/main/packages/http-client would be better idea here.